gamemode/obj_player_extend.lua

```diff
@@ -55,6 +55,75 @@ end
 
 net.Receive('regen_mana', function(len, ply)
   local mana = net.ReadUInt(9)
-  print('regen_mana')
   ply:SetMana(mana)
-end)
\ No newline at end of file
+end)
+
+function meta:GetStatus(type)
+  local ent = self["status_"..type]
+  if ent and ent:GetOwner() == self then 
+    print(ent:GetClass())
+    return ent end
+end
+
+function meta:GiveStatus(type, dur, Effector)
```

Comment:
NIT: Typically function parameters start with a lower case, so `Effector` would be `effector`